src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs

```diff
@@ -232,7 +236,11 @@ private string GenerateCrossgen2ResponseFile()
                 result.AppendLine($"--out:\"{_outputR2RImage}\"");
                 if (!String.IsNullOrEmpty(Crossgen2ExtraCommandLineArgs))
                 {
-                    result.AppendLine(Crossgen2ExtraCommandLineArgs);
+                    foreach (string extraArg in Crossgen2ExtraCommandLineArgs.Split(';'))
```

Comment:
Could we move the for loop after the then-else conditional blocks to remove the duplication now that the code has become a bit more complex?